metagraph/src/cli/transform_annotation.cpp

```diff
@@ -41,6 +41,91 @@ void convert(std::unique_ptr<AnnotatorFrom> annotator,
     target_annotator->serialize(config.outfbase);
 }
 
+template <class T>
+binmat::LinkageMatrix cluster_columns(const std::vector<std::string> &files,
+                                      Config::AnnotationType anno_type,
+                                      uint64_t num_rows_subsampled) {
+    std::vector<uint64_t> row_indexes;
+    std::vector<std::unique_ptr<T>> subcolumn_ptrs;
+    std::vector<uint64_t> column_ids;
+    uint64_t num_rows = 0;
+
+    ThreadPool subsampling_pool(get_num_threads(), 1);
+
+    // Load columns from disk
+    auto on_column = [&](uint64_t i, const std::string &label,
+                         std::unique_ptr<bit_vector> &&column) {
+        subsampling_pool.enqueue([&, i, label, column{std::move(column)}]() {
+            T *subvector;
+            #pragma omp critical
```

Comment:
hmm, I haven't seen `#pragma omp critical` being used with an `std::thread` pool. did you double-check this with a thread sanitizer to make sure it's working as intended?

Reply:
Both of them have the same guarantees. I think externally they are pretty much equivalent.